index.js

```diff
@@ -20,9 +20,10 @@ module.exports = function(haystack, needle, comparator, low, high) {
   }
 
   while(low <= high) {
-    /* Note that "(low + high) >>> 1" may overflow, and results in a typecast
-     * to double (which gives the wrong results). */
-    mid = low + (high - low >> 1);
+    /* low + high >>> 1 could fail for array lengths > 2**31 because >>>
+     * converts its operands to int32. low - (high - low >>> 1) works for
```

Comment:
I believe this line should say "low + (high - low >>> 1)" (that is,  "+" instead of "-")

Reply:
Good catch.